### PR TITLE
Broadcast player results screen

### PR DESCRIPTION
**Pull request by julien4215**
[broadcast_results.webm](https://github.com/user-attachments/assets/bb598fbd-460b-4092-b6c9-85d9975d6772)

### CODE REVIEW

**Comment by ijm8710** [comment]
Hey Julian, 2 requests.

1) mobile web green/reds wins and losses (the 0 and 1s) so that material results stand out. I like this and we should probably do same.
2) on mobile web, above the results list, they have the player bio (ranks across variants, player age, fide id that links to their fide player page,etc). I'd like to have that too shown outright either above or below the list for this pr.

File: lib/src/view/broadcast/broadcast_player_results_screen.dart [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+                    );
+                  }
+
+                  final playerResult = playerResults.games[index - 1];
+
+                  return GestureDetector(
+                    onTap: () {
+                      pushPlatformRoute(
+                        context,
+                        builder: (context) => BroadcastGameScreen(
```
Comment (veloce) [inline lib/src/view/broadcast/broadcast_player_results_screen.dart:227]: Since we can now open a screen from another context, this should be refactored:


https://github.com/lichess-org/mobile/blob/9a6a4004237aac6fac86c87555d7ecf9da7d49dc/lib/src/view/broadcast/broadcast_game_screen.dart#L372-L378

File: lib/src/view/broadcast/broadcast_players_tab.dart [hunk 2]
```diff
@@ -118,7 +121,7 @@ class _PlayersListState extends ConsumerState<PlayersList> {
   @override
   Widget build(BuildContext context) {
     final double eloWidth = max(MediaQuery.sizeOf(context).width * 0.2, 100);
-    final double scoreWidth = max(MediaQuery.sizeOf(context).width * 0.15, 70);
+    final double scoreWidth = max(MediaQuery.sizeOf(context).width * 0.15, 90);
```
Comment (veloce) [inline lib/src/view/broadcast/broadcast_players_tab.dart:124]: Why this change? Have you noticed a case where it was too small?

File: lib/src/view/broadcast/broadcast_players_tab.dart [hunk 3]
```diff
@@ -178,59 +181,78 @@ class _PlayersListState extends ConsumerState<PlayersList> {
...
+                      : player.name,
+                  player.title,
+                  player.name,
+                ),
+              );
+            },
+            child: ColoredBox(
               color: Theme.of(context).platform == TargetPlatform.iOS
-                  ? index.isEven
+                  ? (index - 1).isEven
```
[the remark this answers is not in the document]
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_players_tab.dart:200]: This was to have the same color order as in the opening explorer

File: lib/src/view/broadcast/broadcast_players_tab.dart (hunk 2 above)
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_players_tab.dart:124]: This is because I changed the score from 'x/y' to 'x / y'.

File: lib/src/view/broadcast/broadcast_player_results_screen.dart (hunk 1 above)
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_player_results_screen.dart:227]: I don't understand why this would not work with deep link. The two controllers are independent, they just listen to the same route because there is only one for broadcast.

Comment (veloce) [reply lib/src/view/broadcast/broadcast_player_results_screen.dart:227]: If you can load a game by itself you don't want to load the whole round, right? In a tournament where you have 100 board you'd load 99 boards (and listen to their updates) for nothing.

File: lib/src/view/broadcast/broadcast_boards_tab.dart [hunk 4]
```diff
@@ -60,19 +60,16 @@ class BroadcastBoardsTab extends ConsumerWidget {
...
             child: Center(
               child: Text('Could not load broadcast: $error'),
             ),
           ),
-        _ => const SliverFillRemaining(
-            child: Center(
-              child: CircularProgressIndicator.adaptive(),
-            ),
-          ),
+        _ => const BroadcastPreview.loading()
```
[the remark this answers is not in the document]
Comment (veloce) [reply lib/src/view/broadcast/broadcast_boards_tab.dart:72]: If there is a way to know in advance the number of boards, then we can use the shimmer. Otherwise I think it is ugly to see a flash of 10 shimmer boards where in fact only one is show eventually.

[the remark this answers is not in the document]
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_boards_tab.dart:72]: Having only one board in a round is pretty rare. In the majority of cases I think the board shimmer looks nicer than the circular indicator.

File: lib/src/view/broadcast/broadcast_player_results_screen.dart (hunk 1 above)
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_player_results_screen.dart:227]: I agree with that but I have done it this way because in the API there are no endpoints to get the player information for a specific board. 

Maybe this refactoring can be done in an other branch ?

File: lib/src/view/broadcast/broadcast_game_screen.dart [hunk 5]
```diff
@@ -79,11 +81,15 @@ class _BroadcastGameScreenState extends ConsumerState<BroadcastGameScreen>
   Widget build(BuildContext context) {
     final broadcastGameState = ref
         .watch(broadcastGameControllerProvider(widget.roundId, widget.gameId));
+    final broadcastRoundState =
+        ref.watch(broadcastRoundControllerProvider(widget.roundId));
```
[the remark this answers is not in the document]
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_game_screen.dart]: I used `select` instead because `selectAsync` returns a `Future` which makes unpractical to use inside a widget.

File: lib/src/view/broadcast/broadcast_boards_tab.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (veloce) [reply lib/src/view/broadcast/broadcast_boards_tab.dart]: Unfortunately the world championship is the most watch broadcast... even if it is not the majority it will be seen much more times than other tournaments with more boards.

[the remark this answers is not in the document]
Comment (veloce) [reply lib/src/view/broadcast/broadcast_boards_tab.dart]: It does unless there is only one or two board and you see a flash of shimmer if the request completes very quickly.

File: lib/src/view/broadcast/broadcast_game_screen.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (veloce) [reply lib/src/view/broadcast/broadcast_game_screen.dart]: In order to use `selectAsync` you need to create a dedicated provider to retrieve the title. This is the way to do it with riverpod.

File: lib/src/view/broadcast/broadcast_player_results_screen.dart (hunk 1 above)
Comment (veloce) [reply lib/src/view/broadcast/broadcast_player_results_screen.dart:227]: Yes let's do it in another PR.

Comment (veloce) [reply lib/src/view/broadcast/broadcast_player_results_screen.dart:227]: This code is also problematic because you assume you assume the round controller is loaded: `select((round) => round.requireValue.games[gameId]!)`

I'm not sure it is a good practice to mix `select` and `requireValue` in riverpod. So far we know it is loaded because the only way to access the game screen was through the round screen.

Now that you can access the game screen from the player result screen directly, can you be sure the round screen will be loaded? If you can access a game from another round it won't be the case. This is the main reason why it must be refactored.

Comment (julien4215) [reply lib/src/view/broadcast/broadcast_player_results_screen.dart:228]: I removed the `requireValue`

File: lib/src/view/broadcast/broadcast_game_screen.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_game_screen.dart]: But what is the advantage of creating a new provider instead of using `select` like this
```
final title = ref.watch(
      broadcastRoundControllerProvider(widget.roundId)
          .select((round) => round.value?.round.name),
    );
```

[the remark this answers is not in the document]
Comment (veloce) [reply lib/src/view/broadcast/broadcast_game_screen.dart]: It's mostly a problem of typing. Here you're mixing the async value which represent an asynchronous value which may be not here yet, with a nullable value that you derives from another asynchronous value.

By creating another provider that returns an AsyncValue<String> for the title you're making it clear of what you expect. It is ok to declare a new provider for that. It is like a cached computed value, but async. Doing that is recommended in the riverpod doc (but I don't remember where).

[the remark this answers is not in the document]
Comment (veloce) [reply lib/src/view/broadcast/broadcast_game_screen.dart]: Note also that using `.value?` directly you're not handling the error, it will be rethrown in the widget build method making it crash.

[the remark this answers is not in the document]
Comment (julien4215) [reply lib/src/view/broadcast/broadcast_game_screen.dart]: Ok thanks for the detailed explanation. I'll make a title provider. 

I believe I have already done this in previous PR but I'll make sure to create a provider in the future

File: lib/src/view/broadcast/broadcast_player_results_screen.dart (hunk 1 above)
Comment (veloce) [reply lib/src/view/broadcast/broadcast_player_results_screen.dart:228]: After talking with Thibault, it's decided that it is ok to load the full round to display a single game. So no refactoring after all.

But we need to ensure the round controller is loaded before the game controller in the game screen. (So we can do a `requireValue`)

File: lib/src/view/broadcast/broadcast_game_screen.dart [hunk 6]
```diff
@@ -82,17 +82,21 @@ class _BroadcastGameScreenState extends ConsumerState<BroadcastGameScreen>
...
-          .select((round) => round.value?.round.name),
+    final broadcastGameState = ref.watch(
+      broadcastGameProvider(widget.roundId, widget.gameId),
     );
+    final broadcastGamePgn = ref
+        .watch(broadcastGameControllerProvider(widget.roundId, widget.gameId));
+    final title = widget.title ??
+        (switch (ref.watch(broadcastGameScreenTitleProvider(widget.roundId))) {
+          AsyncData(value: final title) => title,
+          _ => 'Broadcast Game',
```
Comment (veloce) [inline lib/src/view/broadcast/broadcast_game_screen.dart:93]: I'd rather have a `SizedBox.shrink()` here than a default title.

Comment (julien4215) [reply lib/src/view/broadcast/broadcast_game_screen.dart]: Done
